analyses/independent-samples-release/util/independent-dna-samples.R

```diff
@@ -0,0 +1,67 @@
+# independent-dna-samples.R
+
+#' Generate a vector of unique samples
+#' 
+#' The samples from this function will be unique with respect to participants
+#' i.e. only no two samples will come from the same participant. The input list 
+#' should be pre-filtered by `experimental_strategy` and `sample_type`.  
+#' 
+#' 
+#' @param sample_df A data frame of samples, with columns corresponding to those
+#'   in `histologies-base.tsv`
+#' @param tumor_types Designates which types of tumors will be included. Options
+#'   are "primary" to include only primary tumors, "prefer_primary" to include
+#'   primary tumors when available, but fall back to other types, or "any" to
+#'   randomly select among all available specimens. As of v11, primary tumors
+#'   are defined as those designated "Initial CNS Tumor", "Diagnosis", or "Primary Tumor" in the
+#'   `tumor_descriptor` field.
+#' @param seed An optional random number seed. 
+#' 
+#' @return a data frame of Participant and Specimen IDs, each present only once.
+independent_samples <- function(sample_df, 
+                                tumor_types = c("primary", "prefer_primary", "any"), 
+                                seed){
+  tumor_types <- match.arg(tumor_types)
+  if(!missing(seed)){set.seed(seed)}
+  
+  primary_descs <- c("Initial CNS Tumor", "Diagnosis", "Primary Tumor")
+  
+  sample_df <- sample_df %>%
+    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_"))
```

Comment:
```suggestion
    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_sample_id_"))
```

Is it necessary to add the `sample_id_` prefix to prevent the same `match_id` being generated from two distinct `Kids_First_Participant_ID` and `sample_id` combinations. For example:

`Kids_First_Participant_ID` | `sample_id` | `match_id`
--|--|--
`PT_SBP79T7F` | `001` | `PT_SBP79T7F_001`
`PT` | `SBP79T7F_001` | `PT_SBP79T7F_001`

Although there is no such case in v11 histology `histologies-base.tsv`, there is no guarantee for future releases. Adding `"_sample_id_"` prefix makes it very unlikely to have such cases. A complete solution of this issue is to group samples directly using `Kids_First_Participant_ID` and `sample_id`, which may require additional refactoring.

Reply:
@jharenza, do you anticipate at some point in the future distinct  `Kids_First_Participant_ID` and `sample_id` forming a `match_ids` strings that are identical as @logstar has illustrated above?  That is not going the case in the current OpenPedCan datasets.